dts/bindings/interrupt-controller/sifive,plic-1.0.0.yaml

```diff
@@ -12,7 +12,7 @@ properties:
     type: int
     description: Number of external interrupts supported
     required: true
-  riscv,trigger-reg-offset:
-    type: int
-    default: 4224
-    description: Offset of the trigger type register if supported
+
+  support-edge-interrupt:
```

Comment:
Is just a feature that is always available? Do we even need a Devicetree property here? If it's always available, then the Devicetree property is superfluous (and should probably just be removed).

Reply:
@fkokosinski - can you weigh in here?

Reply:
Hi @cfriedt. I've looked around a bit and found out that in the Linux irqchip driver implementation for PLIC SiFive platforms (fu540, fu740; HiFive Unleashed and HiFive Unmatched, respectively) also don't support the edge mode: https://github.com/torvalds/linux/blob/master/drivers/irqchip/irq-sifive-plic.c#L565-L581.

This IMO really suggests that it's not a bug in the documentation that SiFive provides for their chips, and that they really lack the support of edge-triggered interrupts, so I would suggest to  leave the `support-edge-interrupt` property as currently defined by this PR.

Reply:
@keith-packard  - any insider info here from your previous role?

Reply:
@fkokosinski

https://www.kernel.org/doc/Documentation/devicetree/bindings/interrupt-controller/sifive%2Cplic-1.0.0.yaml

Seems to suggest that SiFive's PLIC *does* support edge triggering in the kernel (and that all of them do), and it was signed-off by Paul & Palmer, so I'm inclined to lean toward going with it.

> The PLIC supports both edge-triggered and level-triggered interrupts

What it does say is different in some implementations is whether interrupts are queued or dropped if triggered while they are already pended, which can probably be handled in code, in a somewhat generic way. Good to be aware of, in any case.

Reply:
> @keith-packard - any insider info here from your previous role?

Afraid I'd have to read through documentation I no longer have access to -- I don't specifically remember anything about PLIC interrupt configurations...

Reply:
> Hi @cfriedt. I've looked around a bit and found out that in the Linux irqchip driver implementation for PLIC SiFive platforms (fu540, fu740; HiFive Unleashed and HiFive Unmatched, respectively) also don't support the edge mode: https://github.com/torvalds/linux/blob/master/drivers/irqchip/irq-sifive-plic.c#L565-L581.

Yea, that's the thing - some datasheets do, and some datasheets don't, but the memory map in Linux does support the register, and it's a "quirk" in Linux. Andes and telink support the register and document it in their datasheet. SiFive supports it in their SoC's, but it's undocumented / "reserved", almost as if it were a kind of copy-paste issue from the PLIC spec. 

> SiFive SOCs include an implementation of the Platform-Level Interrupt Controller (PLIC)..  the PLIC supports both edge-triggered and level-triggered interrupts..

http://archive.lwn.net:8080/linux-kernel/b4ddcb20-6109-22ac-4691-c2c012ea112a@wdc.com/T/

I very much suspect that it has been a bit of a contentious topic. The spec was (unintentionally) ambiguous and it's possible that some vendors had one interpretation, while others had another interpretation.

> This IMO really suggests that it's not a bug in the documentation that SiFive provides for their chips, and that they really lack the support of edge-triggered interrupts, so I would suggest to leave the `support-edge-interrupt` property as currently defined by this PR.

https://github.com/torvalds/linux/blob/master/drivers/irqchip/irq-sifive-plic.c#L565-L581.

Of course, we now know that the opposite is true, as we have read. SiFive does, in fact, support edge triggering (via a "quirk"), even though their datasheet does not mention it in the memory map, which is the same "documentation bug" in the PLIC spec itself. Even the ESP32-C3 interrupt controller supports both edge and level-triggered interrupts.

Also important to note, is that `riscv,plic0` and `sifive,plic-1.0.0` are deprecated bindings for "legacy" systems, while all newer compatibles are hard-coded to support edge triggering and use a binding that also hard-codes this support. I would say, that this was a chosen "documentation bug" workaround rather than revising the already-ratified PLIC spec.

https://github.com/torvalds/linux/blob/master/drivers/irqchip/irq-sifive-plic.c#L565-L581

So, IMHO, I think this was a bit of a "documentation bug" that unfortunately caused some misinterpretations of the specification and likely had some undesirable fallout. It happens.

Reply:
The parameter is dropped in #63273